pkg/deploy/deploy.go

```diff
@@ -199,9 +199,6 @@ func deployComponentsParallel(ctx context.Context, components []graph.SortedComp
 	// Collect errors from goroutines and append to the 'errs' slice.
 	for range components {
 		componentDeployErrs := <-errChan
-		if len(componentDeployErrs) > 0 && !opts.ContinueOnErr && !opts.DryRun {
```

Comment:
Random thought: currently, we kinda "hard-cancel" any goroutine still running by the time the CLI execution finished after this return. Probably not what we want to happen anyway